src/NLog/Layouts/JsonLayout.cs

```diff
@@ -116,10 +120,18 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
                 }
             }
 
-            AppendIf(!this.SuppressSpaces, sb, " ");
-            sb.Append("}");
+            var result = sb.ToString();
+
+            if (string.IsNullOrEmpty(result.Trim()) && !RenderEmptyObject)
```

Comment:
Ah, okay! Yeah, not used to thinking performance up front :) Thanks. Will push an update.